src/v/cluster/metrics_reporter.h

```diff
@@ -79,6 +79,9 @@ class metrics_reporter {
         static constexpr int64_t max_size_for_rp_env = 80;
         ss::sstring redpanda_environment;
         ss::sstring id_hash;
+
+        bool has_enterprise_features;
+        bool has_valid_license;
```

Comment:
should these (along with the two bools above) be initialized? i'm a bit surprised that sanitizer hasn't complained before--thought maybe it has and we've missed it. the initialization `metrics_snapshot snapshot;` afaict won't default-value initialize these.

Reply:
ha, i have that change staged locally. same goes for the ints afaik. I'll push it up shortly. 

clang-tidy will only complain if you add `metrics_snapshot() = default;`; not sure whether that's inherent or config-related.